### PR TITLE
upgrade gradle

### LINKED ISSUE #4: Upgrade gradle version
Hello 👋,

Are there any plans in upgrading the gradle version? It is currently blocking the upgrade of some other packages.

I would be more than happy to help with this.

Thanks!

### PR DESCRIPTION
**Pull request by Vlad-Adrian**
This PR is intended to upgrade Gradle plugin version to 8.5 and build tools 8.1.4.

resolves #4

### CODE REVIEW

**Comment by littleGnAl** [comment]
Thanks, @Vlad-Adrian , it is most likely I need to fix the CI first.

**Comment by littleGnAl** [comment]
It's so sorry that this PR https://github.com/littleGnAl/http_proxy_override/pull/6 conflicts with your change, but I think you can safely accept your change when resolving the conflicts.

**Comment by Vlad-Adrian** [comment]
@littleGnAl can you please approve the workflow?

**Comment by littleGnAl** [comment]
I think you need to change the JDK to 17 to pass the CI.
https://github.com/littleGnAl/http_proxy_override/blob/1a950d0df217a848c3611ef78db7c4a82993c2ca/.github/workflows/build.yml#L23

**Comment by littleGnAl** [comment]
After upgrading the gradle version to 8.5, the CI failed with the error:
```
* What went wrong:
Execution failed for task ':http_proxy_override:compileDebugKotlin'.
> Inconsistent JVM-target compatibility detected for tasks 'compileDebugJavaWithJavac' (1.8) and 'compileDebugKotlin' (17).
```
https://github.com/littleGnAl/http_proxy_override/actions/runs/9674481641/job/26690196134?pr=5

It is most likely we need to change the target compatibility to 17, but I'm afraid this will be a breaking change for so many projects. I need to find a solution to avoid breaking it if possible.

**Comment by Vlad-Adrian** [comment]
I am pushing in a few minutes a solution that allowed to build locally at least without java 17

**Comment by littleGnAl** [comment]
@Vlad-Adrian , thanks, I will leave the fix to you.

**Comment by Vlad-Adrian** [comment]
@littleGnAl can you approve the workflow to see if this fixes the issue please?

**Comment by Vlad-Adrian** [comment]
Sorry for the ping pong, I just noticed AGP was commented out. Can you please approve again?

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -25,8 +25,13 @@ apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+kotlin {
```
Comment (littleGnAl) [inline example/android/app/build.gradle:28]: Can you help remove this unnecessary block?

**Comment by littleGnAl** [comment]
@Vlad-Adrian Thanks, it all works now.

**Comment by Vlad-Adrian** [comment]
Thank you too! Will you prepare a new pub release? @littleGnAl

**Comment by littleGnAl** [comment]
I will release it later today.

**Pull request merged**